src/react-components/avatar-selector.js

```diff
@@ -0,0 +1,119 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { injectIntl, FormattedMessage } from 'react-intl';
+
+class AvatarSelector extends Component {
+  static propTypes = {
+    avatars: PropTypes.array,
+    avatar: PropTypes.string,
+    onChange: PropTypes.func,
+  }
+
+  getAvatarIndex(direction=0) {
+    const currAvatarIndex = this.props.avatars.findIndex(avatar => avatar.id === this.props.avatar);
+    const numAvatars = this.props.avatars.length;
+    return ((currAvatarIndex + direction) % numAvatars + numAvatars) % numAvatars;
+  }
+
+  nextAvatar = () => {
+    const newAvatarIndex = this.getAvatarIndex(1);
+    this.props.onChange(this.props.avatars[newAvatarIndex].id);
+  }
+
+  prevAvatar = () => {
+    const newAvatarIndex = this.getAvatarIndex(-1);
+    this.props.onChange(this.props.avatars[newAvatarIndex].id);
+  }
+
+  componentDidUpdate() {
+    // HACK - a-animation ought to restart the animation when the `to` attribute changes, but it doesn't
+    // so we need to force it here.
+    const currRot = this.animation.parentNode.getAttribute('rotation');
+    this.animation.setAttribute('from', `${currRot.x} ${currRot.y} ${currRot.z}`);
+    this.animation.stop();
+    this.animation.handleMixinUpdate();
+    this.animation.start();
+  }
+
+  render () {
+    const avatarAssets = this.props.avatars.map(avatar => (
+      <a-progressive-asset
+        id={avatar.id}
+        key={avatar.id}
+        response-type="arraybuffer"
+        high-src={`./src/assets/avatars/${avatar.models.high}`}
+        low-src={`./src/assets/avatars/${avatar.models.low}`}
+      ></a-progressive-asset>
+    ));
+
+    const avatarEntities = this.props.avatars.map((avatar, i) => (
+      <a-entity key={avatar.id} position="0 0 0" rotation={`0 ${360 * -i / this.props.avatars.length} 0`}>
+        <a-gltf-entity position="0 0 5" rotation="0 0 0" src={'#' + avatar.id} inflate="true">
+          <template data-selector=".RootScene">
+            <a-entity animation-mixer></a-entity>
+          </template>
+          <a-animation
+            attribute="rotation"
+            dur="2000"
+            to={`0 ${this.getAvatarIndex() === i ? 360 : 0} 0`}
+            repeat="indefinite">
+          </a-animation>
+        </a-gltf-entity>
+      </a-entity>
+    ));
+
+    return (
+      <div className="avatar-selector">
+      <span className="avatar-selector__loading">
+        <FormattedMessage id="profile.avatar-selector.loading"/>
+      </span>
+      <a-scene vr-mode-ui="enabled: false" ref={sce => this.scene = sce}>
+        <a-assets>
+          {avatarAssets}
+          <a-asset-item
```

Comment:
we probably should add a issue to eventually create a environment for this and also point this to a gltf bundle url instead of a local asset glb

Reply:
Created #110